Добавлено главное меню

BannikovRuslan opened the pull request.

BannikovRuslan committed:
feat: add main menu

BannikovRuslan committed:
fix: code smell

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/mainmodel/MainScreen.kt:
```diff
@@ -0,0 +1,90 @@
+    )
+    {
+        mainButton(modifier = modifier, fontSize = 126.sp, text = "+")
+        mainButton(modifier = modifier, fontSize = 36.sp, text = "Список")
```
У каждой кнопки будет своя функция ведь.
Еее каждую нужно наверное оформить в виде отдельной функции (это и к кнопке offline относится).

fun addButton
fun listButton
fun onlineButton

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/mainmodel/MainViewModel.kt:
```diff
@@ -0,0 +1,6 @@
+
+
+class MainViewModel {
+    data class MVMState (val isOnline: Boolean)
```
Аббревиатуры нежелательны. Называть как есть MainViewModelState.

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/theme/Theme.kt:
```diff
@@ -22,8 +22,8 @@ private val DarkColorScheme = darkColorScheme(
-        primary = Purple40,
-        secondary = PurpleGrey40,
+        primary = BGLightGrey,
+        secondary = TextBlack,
```
мне кажется тут должны быть конкретные названия переменных, а не общие primary secondary.
Мб поэтому у меня совсем другие цвета на телефоне

BannikovRuslan committed:
fix: исправлены разные недочёты

Цветовая схема в теме приложения по умолчанию задаётся не динамически изменяющейся для новых версий Андроид (12 и выше).

Цветовая схема темы переопределена.

Добавлены пустые функции для кнопок.

Правки структурных элементов меню.

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/mainmodel/MainScreen.kt:
```diff
@@ -0,0 +1,116 @@
+    modifier: Modifier,
+    fontSize: TextUnit,
+    text: String,
+    onClick: () -> Unit,
```
А вот это для чего ?

BannikovRuslan replied:
Параметры функции. Если вопрос про "onClick: () -> Unit", то это функция, связанная с кнопкой, которая ничего не возвращает. [Тут](https://www.bestprog.net/ru/2021/10/07/kotlin-functions-with-a-single-expression-unit-type-ru/#:~:text=%D0%95%D1%81%D0%BB%D0%B8%20%D0%B2%20%D1%82%D0%B5%D0%BB%D0%B5%20%D1%84%D1%83%D0%BD%D0%BA%D1%86%D0%B8%D0%B8%20%D0%BD%D0%B5,%D0%BA%D0%B0%D0%BA%D0%BE%D0%B3%D0%BE%2D%D1%82%D0%BE%20%D1%82%D0%B8%D0%BF%D0%B0%20%D1%8F%D0%B2%D0%BB%D1%8F%D0%B5%D1%82%D1%81%D1%8F%20%D0%BE%D0%B1%D1%8F%D0%B7%D0%B0%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D1%8B%D0%BC) можно посмотреть про Unit.

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/theme/Color.kt:
```diff
@@ -8,4 +8,7 @@ val Pink80 = Color(0xFFEFB8C8)
+val Pink40 = Color(0xFF7D5260)
+
+val BGLightGrey = Color.LightGray
+var TextBlack = Color.Black
```
Наверное эта переменную нужно назвать просто Black. Я так понял в material design не совсем правильно привязываться к конкретным элементам (в данном случае text), есть только первичный, вторичный и т.д цвета.

AntonIV8 commented on src/AndroidODataOfflineSample/app/src/main/java/com/flexberry/androidodataofflinesample/ui/theme/Theme.kt:
```diff
@@ -41,7 +43,7 @@ private val LightColorScheme = lightColorScheme(
         darkTheme: Boolean = isSystemInDarkTheme(),
         // Dynamic color is available on Android 12+
-        dynamicColor: Boolean = true,
+        dynamicColor: Boolean = false,
```
Добавить коммент

BannikovRuslan committed:
fix: добавлены исправления по замечаниям

Добавлен комментарий в AndroidODataOfflineSampleTheme.

Изменены имена переменных, связанные с цветами темы.

Внесены правки по Code Smell
